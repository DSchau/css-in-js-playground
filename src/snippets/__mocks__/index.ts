import * as fs from 'fs';
import * as path from 'path';

const base = path.join(__dirname, '../');
const ignore = ['.DS_Store', '__tests__', '__mocks__'];

const allFilesAndFolders = fs
  .readdirSync(base)
  .filter(fileOrDirectory => {
    if (ignore.indexOf(fileOrDirectory) > -1) {
      return false;
    }
    return fs.statSync(path.join(base, fileOrDirectory)).isDirectory();
  })
  .map(directory => {
    const snippets = fs
      .readdirSync(path.join(base, directory))
      .filter(file => ignore.indexOf(file) === -1);
    return [directory, snippets];
  }) as string[][];

const files = allFilesAndFolders
  .slice(0, 1)
  .shift()
  .slice(1)
  .pop();

export { files };

export default allFilesAndFolders.reduce((libraries, [folder, files]) => {
  libraries[folder] = [].concat(files).reduce((library, file) => {
    library[file] = fs.readFileSync(path.join(base, folder, file), 'utf8');
    return library;
  }, {});
  return libraries;
}, {});
