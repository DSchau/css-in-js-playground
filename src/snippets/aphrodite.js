import React, { Component } from 'react';
import { StyleSheet, css } from 'aphrodite';
import Logo from './logo';

class Login extends Component {
  constructor(props) {
    super(props);
    this.state = {
      email: '',
      phoneNumber: '',
      fields: ['email', 'phoneNumber'],
      valid: false
    };
  }

  handleInputChange() {
    return ev => {
      const { value } = ev.target;
      const prop = ev.target.getAttribute('name');
      this.setState({
        [prop]: value,
        valid:
          value.length > 0 &&
          this.state.fields.every(field => this.state[field].length > 0)
      });
    };
  }

  render() {
    return (
      <div className={css(styles.container)}>
        <header className={css(styles.header)}>
          <div className={css(styles.logoContainer)}>
            <Logo color="#25b57f" size={32} />
          </div>
          <div className={css(styles.stripes)}>
            <div className={css(styles.stripeBottom)} />
          </div>
          <div className={css(styles.titleContainer)}>
            <h1 className={css(styles.title)}>Set up your payments</h1>
            <h1 className={css(styles.title, styles.subTitle)}>
              Rocketship, Inc.
            </h1>
          </div>
        </header>
        <div className={css(styles.stripe)} />
        <div className={css(styles.form)} onSubmit={ev => ev.preventDefault()}>
          <input
            type="text"
            name="email"
            className={css(styles.input)}
            placeholder="Email"
            onChange={this.handleInputChange()}
          />
          <input
            type="text"
            name="phoneNumber"
            className={css(styles.input)}
            placeholder="Phone number"
            onChange={this.handleInputChange()}
          />
          <button
            type="submit"
            className={css(
              ...[
                styles.submitButton,
                this.state.valid
                  ? styles.submitButtonEnabled
                  : styles.submitButtonDisabled
              ]
            )}
            disabled={!this.state.valid}
          >
            Submit
          </button>
        </div>
      </div>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    display: 'flex',
    flexDirection: 'column',
    minHeight: '100%',
    width: '100%',
    backgroundColor: '#f6f9fc'
  },
  header: {
    display: 'flex',
    flexDirection: 'column',
    height: '40%',
    minHeight: 250,
    width: '100%',
    position: 'relative',
    justifyContent: 'center'
  },
  logoContainer: {
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
    height: 60,
    width: 60,
    borderRadius: 60,
    backgroundColor: 'white',
    boxShadow: '0 1px 6px rgba(0, 0, 0, 0.1)',
    position: 'absolute',
    top: '1rem',
    left: '1rem',
    zIndex: 2
  },
  stripes: {
    position: 'absolute',
    top: 0,
    right: 0,
    bottom: 0,
    left: 0,
    width: '100%',
    height: '100%',
    overflow: 'hidden',
    transform: 'skewY(-8deg)',
    transformOrigin: 0,
    background: 'linear-gradient(-150deg, #25b57f 0%, #acf0b5 70%)'
  },
  stripe: {
    height: '10vh',
    overflow: 'hidden',
    transform: 'skewY(-8deg)',
    transformOrigin: 0,
    background:
      'linear-gradient(-150deg, rgba(255, 255, 255, 0) 40%, #ddecf7 70%)'
  },
  stripeBottom: {
    position: 'absolute',
    bottom: 0,
    left: 0,
    right: 0,
    height: '35%',
    overflow: 'hidden',
    background:
      'linear-gradient(150deg, rgba(172, 240, 181, 0) 40%, #acf0b5 70%)'
  },
  titleContainer: {
    padding: '1rem'
  },
  title: {
    color: 'white',
    margin: 0,
    padding: 0,
    position: 'relative',
    fontSize: '1.5rem',
    fontWeight: '100'
  },
  subTitle: {
    color: '#025450',
    marginTop: '0.5rem'
  },
  form: {
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center',
    justifyContent: 'center',
    width: '100%',
    padding: '2rem',
    boxSizing: 'border-box'
  },
  input: {
    display: 'block',
    width: '100%',
    marginBottom: '1rem',
    padding: '1.25rem 1rem',
    boxSizing: 'border-box',
    borderRadius: '0.25rem',
    border: '1px solid transparent',
    boxShadow: '0 1px 6px rgba(0, 0, 0, 0.1)',
    ':focus': {
      borderColor: '#6772e5',
      outline: 'none',
      boxShadow: '0 1px 6px rgba(103, 114, 229, 0.5)'
    }
  },
  submitButton: {
    display: 'block',
    color: 'white',
    border: 'none',
    width: '100%',
    padding: '1.25rem 1rem',
    boxSizing: 'border-box',
    borderRadius: '0.25rem',
    textTransform: 'uppercase',
    boxShadow: '0 1px 6px rgba(0, 0, 0, 0.1)',
    marginTop: '1rem'
  },
  submitButtonEnabled: {
    backgroundColor: '#6772e5'
  },
  submitButtonDisabled: {
    backgroundColor: '#BBB'
  }
});

export default Login;
