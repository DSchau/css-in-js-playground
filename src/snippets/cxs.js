import React, { Component } from 'react';
import cxs from 'cxs/component';
import Logo from './logo';

const Container = cxs('main')({
  display: 'flex',
  flexDirection: 'column',
  minHeight: '100%',
  width: '100%',
  backgroundColor: '#f6f9fc'
});

const Header = cxs('header')({
  display: 'flex',
  flexDirection: 'column',
  height: '40%',
  minHeight: '250px',
  width: '100%',
  position: 'relative',
  justifyContent: 'center'
});

const LogoContainer = cxs('div')({
  display: 'flex',
  alignItems: 'center',
  justifyContent: 'center',
  height: '60px',
  width: '60px',
  borderRadius: '60px',
  backgroundColor: 'white',
  boxShadow: '0 1px 6px rgba(0, 0, 0, 0.1)',
  position: 'absolute',
  top: '1rem',
  left: '1rem',
  zIndex: 2
});

const Stripes = cxs('div')({
  position: 'absolute',
  top: 0,
  right: 0,
  bottom: 0,
  left: 0,
  width: '100%',
  height: '100%',
  overflow: 'hidden',
  transform: 'skewY(-8deg)',
  transformOrigin: 0,
  background: 'linear-gradient(-150deg, #25b57f 0%, #acf0b5 70%)'
});

const Stripe = cxs('div')({
  height: '10vh',
  overflow: 'hidden',
  transform: 'skewY(-8deg)',
  transformOrigin: 0,
  background:
    'linear-gradient(-150deg, rgba(255, 255, 255, 0) 40%, #ddecf7 70%)'
});

const StripeBottom = cxs('div')({
  position: 'absolute',
  bottom: 0,
  left: 0,
  right: 0,
  height: '35%',
  overflow: 'hidden',
  background: 'linear-gradient(150deg, rgba(172, 240, 181, 0) 40%, #acf0b5 70%)'
});

const TitleContainer = cxs('div')({
  padding: '1rem'
});

const Title = cxs('h1')({
  color: 'white',
  margin: 0,
  padding: 0,
  position: 'relative',
  fontSize: '1.5rem',
  fontWeight: '100'
});

const SubTitle = cxs(Title)({
  color: '#025450',
  marginTop: '0.5rem'
});

const Form = cxs('form')({
  display: 'flex',
  flexDirection: 'column',
  alignItems: 'center',
  justifyContent: 'center',
  width: '100%',
  padding: '2rem',
  boxSizing: 'border-box',
  zIndex: '2'
});

const Input = cxs('input')({
  display: 'block',
  width: '100%',
  marginBottom: '1rem',
  padding: '1.25rem 1rem',
  boxSizing: 'border-box',
  borderRadius: '0.25rem',
  border: '1px solid transparent',
  boxShadow: '0 1px 6px rgba(0, 0, 0, 0.1)',
  ':focus': {
    borderColor: '#6772e5',
    outline: 'none',
    boxShadow: '0 1px 6px rgba(103, 114, 229, 0.5)'
  }
});

const SubmitButton = cxs('button')(props => ({
  display: 'block',
  backgroundColor: props.disabled ? '#BBB' : '#6772e5',
  color: 'white',
  border: 'none',
  width: '100%',
  padding: '1.25rem 1rem',
  boxSizing: 'border-box',
  borderRadius: '0.25rem',
  textTransform: 'uppercase',
  boxShadow: '0 1px 6px rgba(0, 0, 0, 0.1)',
  marginTop: '1rem'
}));

SubmitButton.defaultProps = {
  type: 'submit'
};

export default class Login extends Component {
  constructor(props) {
    super(props);
    this.state = {
      email: '',
      phoneNumber: '',
      fields: ['email', 'phoneNumber'],
      valid: false
    };
  }

  handleInputChange() {
    return ev => {
      const { value } = ev.target;
      const prop = ev.target.getAttribute('name');
      this.setState({
        [prop]: value,
        valid:
          value.length > 0 &&
          this.state.fields.every(field => this.state[field].length > 0)
      });
    };
  }

  render() {
    return (
      <Container>
        <Header>
          <LogoContainer>
            <Logo color="#25b57f" size={32} />
          </LogoContainer>
          <Stripes>
            <StripeBottom />
          </Stripes>
          <TitleContainer>
            <Title>Set up your payments</Title>
            <SubTitle>Rocketship, Inc.</SubTitle>
          </TitleContainer>
        </Header>
        <Stripe />
        <Form onSubmit={ev => ev.preventDefault()}>
          <Input
            type="text"
            name="email"
            placeholder="Email"
            onChange={this.handleInputChange()}
          />
          <Input
            type="text"
            name="phoneNumber"
            placeholder="Phone number"
            onChange={this.handleInputChange()}
          />
          <SubmitButton disabled={!this.state.valid}>Submit</SubmitButton>
        </Form>
      </Container>
    );
  }
}
