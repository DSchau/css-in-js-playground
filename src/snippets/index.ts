const context = require.context('.', true, /\.js$/);
const keys = context.keys();

const lib = keys.reduce((lib, path) => {
  let [, library, fileName] = path.split('/');
  const [file] = fileName.split('.js');
  lib[library] = {
    ...(lib[library] || {}),
    [file]: context(path)
  };
  return lib;
}, {});

const { libraries, files } = keys.reduce(
  (lib, path) => {
    let [, library, fileName] = path.split('/');
    const [file] = fileName.split('.js');
    if (lib.files.indexOf(file) === -1) {
      lib.files.push(file);
    }
    lib.libraries[library] = {
      ...(lib.libraries[library] || {}),
      [file]: context(path)
    };
    return lib;
  },
  {
    libraries: {},
    files: []
  }
);

export { files };

export default libraries;
