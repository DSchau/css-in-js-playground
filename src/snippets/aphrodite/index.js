import React from 'react';
import { StyleSheet, css } from 'aphrodite';

import Form from './form';
import Header from './header';

function Login() {
  return (
    <div className={css(styles.container)}>
      <Header />
      <div className={css(styles.stripe)} />
      <Form />
    </div>
  );
}

const styles = StyleSheet.create({
  container: {
    display: 'flex',
    flexDirection: 'column',
    minHeight: '100%',
    width: '100%',
    backgroundColor: '#f6f9fc'
  },
  stripe: {
    height: '10vh',
    overflow: 'hidden',
    transform: 'skewY(-8deg)',
    transformOrigin: 0,
    background:
      'linear-gradient(-150deg, rgba(255, 255, 255, 0) 40%, #ddecf7 70%)'
  }
});

export default Login;
