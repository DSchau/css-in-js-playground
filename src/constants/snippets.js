import * as StyledComponents from '../snippets/styled-components';
import * as Glamorous from '../snippets/glamorous';
import * as Aphrodite from '../snippets/aphrodite';
import * as ReactJSS from '../snippets/react-jss';
import * as JSS from '../snippets/jss';
import * as CXS from '../snippets/cxs';
import * as Radium from '../snippets/radium';

export { StyledComponents, Glamorous, Aphrodite, ReactJSS, JSS, CXS, Radium };
