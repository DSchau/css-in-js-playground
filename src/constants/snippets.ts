import * as StyledComponents from '../snippets/styled-components';
import * as Glamorous from '../snippets/glamorous';
import * as Aphrodite from '../snippets/aphrodite';
import * as Cxs from '../snippets/cxs';
import * as Radium from '../snippets/radium';
import * as ReactJSS from '../snippets/react-jss';
import * as JSS from '../snippets/jss';

export { StyledComponents, Glamorous, Aphrodite, Cxs, Radium, ReactJSS, JSS };
