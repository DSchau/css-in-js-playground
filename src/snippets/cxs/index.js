import React from 'react';
import cxs from 'cxs/component';

import Form from './form';
import Header from './header';

const Container = cxs('main')({
  display: 'flex',
  flexDirection: 'column',
  minHeight: '100%',
  width: '100%',
  backgroundColor: '#f6f9fc'
});

const Stripe = cxs('div')({
  height: '10vh',
  overflow: 'hidden',
  transform: 'skewY(-8deg)',
  transformOrigin: 0,
  background:
    'linear-gradient(-150deg, rgba(255, 255, 255, 0) 40%, #ddecf7 70%)'
});

export default function Login() {
  return (
    <Container>
      <Header />
      <Stripe />
      <Form />
    </Container>
  );
}
