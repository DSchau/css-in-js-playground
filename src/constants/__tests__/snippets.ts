import * as SNIPPETS from '../snippets';

test('it exports common libraries', () => {
  const keys = Object.keys(SNIPPETS).sort();

  expect(keys).toMatchSnapshot();
});
