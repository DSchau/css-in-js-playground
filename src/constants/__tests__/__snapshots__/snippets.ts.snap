// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it exports common libraries 1`] = `
Array [
  "Aphrodite",
  "Cxs",
  "Glamorous",
  "JSS",
  "Radium",
  "ReactJSS",
  "StyledComponents",
]
`;
